crates/edr_evm/src/chain_spec.rs

```diff
@@ -0,0 +1,46 @@
+use std::fmt::Debug;
+
+use edr_eth::{transaction::SignedTransaction, Address};
+use revm::primitives::TxEnv;
+
+/// A trait for defining a chain's associated types.
+pub trait ChainSpec {
+    /// The type of signed transactions used by this chain.
+    type SignedTransaction: alloy_rlp::Encodable
+        + Clone
+        + Debug
+        + IntoTxEnv
+        + PartialEq
+        + Eq
+        + SignedTransaction;
+}
+
+/// The chain specification for Ethereum Layer 1.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct L1ChainSpec;
+
+impl ChainSpec for L1ChainSpec {
+    type SignedTransaction = edr_eth::transaction::Signed;
+}
+
+// TODO: This is a temporary solution until the revm fork has been merged. In
+// the fork each chain type has its own transaction type with an accompanying
+// trait.
```

Comment:
Nice to have: if there is an issue/PR for this it'd be good to link here.

Reply:
Addressed in https://github.com/NomicFoundation/edr/pull/493/commits/ca51139abae53f862413947ab1b789b95550b2b5